libs/spark/src/Unstable_CheckboxListItem/Unstable_CheckboxListItem.tsx

```diff
@@ -0,0 +1,205 @@
+  classKey: Unstable_CheckboxListItemClassKey;
+
+  // 'button' | 'onClick' | 'primaryAction' | 'value'
+  props: P & {
+    /**
+     * Props applied to the `Checkbox` element.
+     */
+    CheckboxProps?: Partial<Unstable_CheckboxProps>;
+    /**
+     * If `true`, the component is checked.
+     */
+    checked?: Unstable_CheckboxProps['checked'];
+    /**
+     * If `true`, the component appears indeterminate.
+     */
+    indeterminate?: Unstable_CheckboxProps['indeterminate'];
+    /**
+     * A `classes` prop applied to the `ListItem` element.
+     */
+    ListItemClasses?: Unstable_ListItemProps['classes'];
+    /**
+     * Callback fired when the component is clicked (either the checkbox input or list item).
+     *
+     * @param {object} event The event source of the callback.
+     * @param {string} value The current value of the component
+     * @param {boolean} checked The current checked state of the component
+     */
+    onClick?: (
+      // eslint-disable-next-line @typescript-eslint/ban-types
+      event: MouseEvent<{}>,
+      value: string,
+      checked: boolean
+    ) => void;
+    /**
+     * An alias for the `checked` prop.
+     */
+    selected?: Unstable_ListItemProps['selected'];
+    /**
+     * The value of the component.
+     */
+    value: string;
+  };
+};
+
+export type Unstable_CheckboxListItemProps<
+  D extends ElementType = Unstable_CheckboxListItemTypeMap['defaultComponent'],
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  P = {}
+> = OverrideProps<Unstable_CheckboxListItemTypeMap<P, D>, D>;
+
+export type Unstable_CheckboxListItemClassKey = 'root' | 'primary';
+
+const useStyles = makeStyles(
+  {
+    root: {},
+    primary: {
+      '$root:hover &': {
+        textDecoration: 'underline',
+      },
+    },
+  },
+  { name: 'MuiSparkUnstable_CheckboxListItem' }
+);
+
+// @ts-expect-error can't handle overloads by `button` values
+const Unstable_CheckboxListItem: OverridableComponent<
+  Unstable_CheckboxListItemTypeMap<
+    { button: false },
+    Unstable_CheckboxListItemTypeMap['defaultComponent']
+  >
+> &
+  ExtendButtonBase<
+    Unstable_CheckboxListItemTypeMap<
+      { button?: true },
+      Unstable_CheckboxListItemTypeMap['defaultComponent']
+    >
+  > = forwardRef(function Unstable_CheckboxListItem(props, ref) {
+  const {
+    className,
+    CheckboxProps,
+    checked: checkedProp,
+    children,
+    classes: classesProp,
+    disabled,
+    id: idProp,
+    indeterminate,
+    ListItemClasses,
+    onClick,
+    primary,
+    primaryTypographyProps,
+    secondary,
+    secondaryTypographyProps,
+    selected,
+    tertiary,
+    tertiaryTypographyProps,
+    value,
+    ...other
+  } = props;
+
+  const classes = useStyles();
+
+  const primaryId = useId(primaryTypographyProps?.id);
+  const secondaryId = useId(secondaryTypographyProps?.id);
+  const tertiaryId = useId(tertiaryTypographyProps?.id);
+
+  if (process.env.NODE_ENV !== 'production') {
```

Comment:
This is new, why add this?

Reply:
I think it's in some other components. But its to warn consumers in development when they're using it wrong but not in production.